app/models/event.rb

```diff
@@ -0,0 +1,71 @@
+class Event < ApplicationRecord
+  # extends ...................................................................
+
+  # includes ..................................................................
+  include Taggable
+
+  # relationships .............................................................
+  belongs_to :eventable, polymorphic: true
+  belongs_to :user
+
+  # validations ...............................................................
+  validates :body, presence: true
+  validates :user, presence: true
+
+  # callbacks .................................................................
+  # scopes ....................................................................
+
+  # Helper class method to lookup all events assigned
+  # to all eventable types for a given user.
+  scope :find_events_by_user, ->(user) { where(user_id: user.id).order("created_at DESC") }
+
+  # Helper class method to look up all events for
+  # eventable class name and eventable id.
+  scope :find_events_for_eventable, ->(eventable_str, eventable_id) do
+    where(eventable_type: eventable_str.to_s, eventable_id: eventable_id).order("created_at DESC")
+  end
+
+  # Scopes and helpers provied by tag_columns
+  # SEE: https://github.com/hopsoft/tag_columns
+  #
+  # - with_tags
+  # - without_tags
+  # - with_any_tags
+  # - without_any_tags
+  # - with_all_tags
+  # - without_all_tags
+  #
+  # Examples
+  #
+  #   irb>User.events.with_tags("registration", "email")
+
+  # additional config (i.e. accepts_nested_attribute_for etc...) ..............
+  tag_columns :tags
+
+  # class methods .............................................................
+  class << self
+    alias by_user find_events_by_user
+    alias for_eventable find_events_for_eventable
```

Comment:
Do we need aliases here or can this simply be the scope name?

Reply:
good point